src/client/components/Header/Header.tsx

```diff
@@ -1,14 +1,144 @@
 import React, { Component } from 'react';
+import './Header.css';
 
-export class Header extends Component {
-    constructor(props: any) {
-      super(props);
-      this.state = {};
+import {fetchProfileAPI} from '../../utils/API';
+import { getUsername } from '../../utils/auth';
+
+interface State {
+  dropdownHidden: boolean;
+  gravatarHash: string | undefined;
```

Comment:
Instead of `gravatarHash: string | undefined` can't it be `gravatarHash?: string;`?